MyLibrary/Tests/MyLibraryTests/MyLibraryTests.swift

```diff
@@ -68,5 +69,38 @@ final class MyLibraryTests: XCTestCase {
         // Then
         XCTAssertNil(isLuckyNumber)
     }
+    
+    func testWeatherModelForValidTemperature() {
+        let decoder = JSONDecoder()
+        let data = Data("{\"main\": {\"temp\":58}}".utf8)
+        do {
+            let container = try decoder.decode(Weather.self, from: data)
+            XCTAssertNotNil(container)
+            XCTAssert(container.main.temp == 58.0)
+        } catch {
+            XCTFail("Failed to decode temp from json")
+        }
+    }
+    func testWeatherModelForInvalidTemperature() {
+        let decoder = JSONDecoder()
+        let data = Data("{\"main\": {\"temp\":\"invalid temp\"}}".utf8)
+        do {
+            let _ = try decoder.decode(Weather.self, from: data)
+            XCTFail("Should not have been possible to decode temp for invalid input")
+        } catch {
+            XCTAssert(true == true)
+        }
+    }
+    
+    // Integration Tests
+    func testWeatherServiceReturnsCorrectTemperature() async {
+        let weatherServiceObj = WeatherServiceImpl()
+        do{
+            let temp = try await weatherServiceObj.getTemperature()
+            XCTAssert(temp is Int)
+        } catch {
+            XCTFail("Failed to hit weather service")
+        }
+    }
```

Comment:
I would add some tests for the model too.  That way, if the model breaks (e.g. someone puts a PR that breaks it), you'll know right away.